### PR TITLE
Refactor/plugin iterator

### PR DESCRIPTION
**Pull request by Tieske**

### CODE REVIEW

**Comment by Tieske** [comment]
locally all tests pass

File: kong/core/plugins_iterator.lua [hunk 1]
```diff
@@ -52,18 +52,16 @@ local function iter_plugins_for_req(loaded_plugins, is_access_or_certificate_con
   end
 
   local i = 0
-  local function get_next_plugin()
-    i = i + 1
-    return loaded_plugins[i]
-  end
+  local empty = {}
```
Comment (thibaultCha) [inline kong/core/plugins_iterator.lua]: We could just create it on the chunk level, to avoid creating a new table on each new iterator.

Comment (Tieske) [inline kong/core/plugins_iterator.lua]: done!

**Comment by Tieske** [comment]
will push new against `refactor/cli` branch

**Pull request closed without merging**